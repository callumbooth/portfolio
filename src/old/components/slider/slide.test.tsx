import React from 'react';

import { render } from '@testing-library/react';

import Slide from './slide';

it('Slide renders correctly', () => {
    const data = {
        name: 'Super theme',
        slug: 'super-theme',
        rotate: 0,
        featured: true,
        tags: ['tag', 'tag2', 'tag3'],
        launched: 'September 2018',
        summary: 'A Moodle theme to end all Moodle themes',
        content: 'content goes here',
    };

    const component = render(
        <Slide
            delayTransition={jest.fn()}
            loaded={true}
            data={data}
            toggleInfo={jest.fn()}
            currentSlide={1}
        />,
    );

    expect(component.container).toMatchSnapshot();
});
