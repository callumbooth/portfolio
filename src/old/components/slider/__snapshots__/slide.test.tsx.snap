// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide renders correctly 1`] = `
<div>
  <div
    class="w-full h-full absolute"
    style="opacity: 0;"
  >
    <div
      class="background-wrapper-top absolute w-full top-0 z-0 flex justify-start overflow-hidden"
    >
      <svg
        class="flex-grow w-full h-full"
        viewBox="0 0 1520 1080"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <line
            class="stroke-current text-primary-main stroke-3"
            fill="none"
            pathLength="100"
            stroke-dasharray="100"
            stroke-dashoffset="100"
            transform="translate(1090 0) rotate(0 217.5 217.5)"
            x2="435"
            y2="435"
          />
          <line
            class="stroke-current text-primary-main stroke-3"
            fill="none"
            pathLength="100"
            stroke-dasharray="100"
            stroke-dashoffset="100"
            transform="translate(417 0) rotate(0 184.5 184.5)"
            x2="369"
            y2="369"
          />
        </g>
      </svg>
    </div>
    <div
      class="slide-wrapper flex flex-wrap h-full items-center justify-center relative z-10"
      style="opacity: 0;"
    >
      <div
        class="w-full max-w-full font-bold text-primary-main text-center text-6xl leading-none p-7 lg:text-8xl lg:p-10 xl:text-9xl xl:p-12"
      >
        <a
          href="/projects/super-theme"
        >
          Super theme
        </a>
      </div>
      <div
        class="absolute bottom-12 left-12"
      >
        <button
          class="btn btn-red"
        >
          i
        </button>
        <div
          class="absolute w-72 left-1 bottom-full opacity-0 p-5 rounded-sm bg-primary-main text-white"
        >
          <p>
            A Moodle theme to end all Moodle themes
          </p>
          <a
            class="project-link uppercase text-white text-xs"
            href="/projects/super-theme"
          >
            View Project
            <svg
              class="inline-block w-4"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      class="transition duration-500 absolute w-full bottom-0 z-0 flex justify-start overflow-hidden"
    >
      <svg
        class="flex-grow w-full h-full"
        viewBox="0 0 1520 1080"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            class=" fill-current text-gray-200"
            d="M0,0 l919,919 h-218 l-701,-701 Z"
            opacity="0"
            transform="translate(0 190) rotate(0 459.5 459.5)"
          />
          <line
            class="stroke-current text-primary-main stroke-3"
            fill="none"
            pathLength="100"
            stroke-dasharray="100"
            stroke-dashoffset="100"
            transform="translate(0 410) rotate(0 350.5 350.5)"
            x2="701"
            y2="701"
          />
          <line
            class="stroke-current text-primary-main stroke-3"
            fill="none"
            pathLength="100"
            stroke-dasharray="100"
            stroke-dashoffset="100"
            transform="translate(0 190) rotate(0 460.5 460.5)"
            x2="921"
            y2="921"
          />
          <line
            class="stroke-current text-primary-main stroke-3"
            fill="none"
            pathLength="100"
            stroke-dasharray="100"
            stroke-dashoffset="100"
            transform="translate(1086 669) rotate(0 205.5 205.5)"
            x2="411"
            y2="411"
          />
          <line
            class="stroke-current text-primary-main stroke-3"
            fill="none"
            pathLength="100"
            stroke-dasharray="100"
            stroke-dashoffset="100"
            transform="translate(1008 873)"
            x2="207"
            y2="207"
          />
        </g>
      </svg>
    </div>
  </div>
</div>
`;
